packages/cli/internal/pkg/cli/context/info.go

```diff
@@ -4,8 +4,11 @@ import (
 	"reflect"
 
 	"github.com/aws/amazon-genomics-cli/internal/pkg/cli/spec"
+	"github.com/aws/amazon-genomics-cli/internal/pkg/constants"
 )
 
+var headEngines = map[string]bool{constants.NEXTFLOW: true, constants.MINIWDL: true, constants.SNAKEMAKE: true}
```

Comment:
From maintainability point of view, all future engines most likely will be "HeadEngine". It means every time we add a new engine we will have to update this line (easy to forget). It would be more reliable if we do the opposite. We can store only one (legacy) engine which is NOT a Head Engine. This way our implementation will be future-proof.

Reply:
Good idea. I'll keep it as a map because I think Toil will be a server engine as well but these are the exception rather than the rule.